.github/workflows/changesets.yml

```
@@ -21,19 +18,30 @@ jobs:
               with:
                   fetch-depth: 0
 
+            - name: setup pnpm
+              uses: pnpm/action-setup@v4
+
             - name: setup node.js
               uses: actions/setup-node@v3
               with:
                   node-version: 20
-            - name: install pnpm
-              run: npm i pnpm@latest -g
+
+            # PNPM Store cache setup
+            - name: get pnpm store directory
+              id: pnpm-cache
+              run: |
+                  echo "pnpm_cache_dir=$(pnpm store path)" >> $GITHUB_OUTPUT
```

do we need this part? I'm not quite sure what it does, but it's not present in pnpm's docs (https://github.com/pnpm/action-setup) and it seems like it'd save a lot of lines to remove it wherever it shows up
```suggestion
            - name: get pnpm store directory
              id: pnpm-cache
              run: |
                  echo "pnpm_cache_dir=$(pnpm store path)" >> $GITHUB_OUTPUT
```

---

It's there, at the bottom of the README here: https://github.com/pnpm/action-setup#use-cache-to-reduce-installation-time

In the past, I've found using https://github.com/actions/cache directly to be far more reliable than using the built-in `cache` option in https://github.com/actions/setup-node. Perhaps it's better nowadays? I'd be happy to give it another go

---

oh, their setup is a bit different. e.g. they use `GITHUB_ENV` instead of `GITHUB_OUTPUT`, which is why I didn't find it when I did "Ctrl+F" for "GITHUB_OUTPUT"

anyway, we've been using the built-in `cache: 'pnpm'` in SvelteKit for ages and I haven't noticed any issues with it, but I don't know what issues you've seen

---

> but I don't know what issues you've seen

back when i was setting it up for skeleton (probably a little over a year ago now) it wasn't actually caching for some reason 🤷

it looks like it's working here though!